web-client/src/presenter/actions/performanceMeasurementEndAction.ts

```diff
@@ -0,0 +1,34 @@
+import { DateTime } from 'luxon';
+
+export const performanceMeasurementEndAction = ({
+  applicationContext,
+  props,
+}: ActionProps<{
+  sequenceName?: string;
+  performanceMeasurementStart?: number;
+  actionPerformanceArray?: { actionName: string; duration: number }[];
+}>) => {
+  const { actionPerformanceArray, performanceMeasurementStart, sequenceName } =
+    props;
+
+  if (!sequenceName || !performanceMeasurementStart || !actionPerformanceArray)
+    return;
+
+  const performanceMeasurementEnd = DateTime.now().toMillis();
+  const durationInSeconds =
+    (performanceMeasurementEnd - performanceMeasurementStart) / 1000;
+
+  const RESULTS: {
+    sequenceName: string;
+    duration: number;
+    actionPerformanceArray: { actionName: string; duration: number }[];
+  } = {
+    actionPerformanceArray,
+    duration: durationInSeconds,
+    sequenceName,
+  };
+
+  applicationContext
+    .getUseCases()
+    .logUserPerformanceDataInteractor(applicationContext, RESULTS);
```

Comment:
Just glancing over things, so I might be missing something--what if this call fails? Especially since we are using this "everywhere," I'm thinking we probably want to have better exception handling so the app works even if our performance monitoring doesn't.

Reply:
Nevermind, ignore.